src/engine/server/sv_client.cpp

```diff
@@ -673,7 +673,7 @@ when the cvar is set to something, the download server will effectively never us
 */
 static bool SV_CheckFallbackURL( client_t *cl, const char* pakName, msg_t *msg )
 {
-	if ( !sv_wwwFallbackURL->string || strlen( sv_wwwFallbackURL->string ) == 0 )
+	if ( !sv_wwwFallbackURL->string || !*sv_wwwFallbackURL->string )
```

Comment:
I'm not very happy with this kind of `!!*!*!!!*!` optimization that affects readability. Is there any alternate option?
If it were me I would replace existing `!(non bool)` tests with explicit tests against `0`, `\0`, `NULL` or `nullptr`…

Reply:
`if (a == nullptr || a[0] == '\0')`?

Also may be refactored to a function, like `bool is_null_or_empty(const char*) `

Reply:
I like the `a[0] == '\0'` idea, I would like to know if @slipher has an opinion or something against this.

At least, this sounds less obscure to me (if I'm not doing mistake, it's equivalent and it works):

```c++
if ( !sv_wwwFallbackURL->string || !sv_wwwFallbackURL->string[0] )
```

This would cut it in half: not too verbose, explicit enough. That requires less understanding of pointer mechanism to understand what's done, at least it's easy to guess the right part is testing the content. Even a python programmer who would have no clue about c and pointers can infer `if None or if empty string`. This would encourage contributors to write good code the cargo cult way. =D

I would not be against a `is_null_or_empty` function if there is more than one or two occurrences.

Reply:
@dimhotepus I guess `!sv_wwwFallbackURL->string[0]` and `!*sv_wwwFallbackURL->string` produces the same code, right?

Reply:
```
if ( !sv_wwwFallbackURL->string || !sv_wwwFallbackURL->string[0] )
```

Reply:
Yes, if you do this it would look OK to me. Then also squash the fix for the loaded pack reference and I'll merge the PR.